src/status_im/contexts/wallet/wallet_connect/events/session_responses.cljs

```diff
@@ -6,96 +6,84 @@
              data-store]
             [status-im.contexts.wallet.wallet-connect.utils.uri :as uri]
             [taoensso.timbre :as log]
+            [utils.hex :as hex]
             [utils.i18n :as i18n]
             [utils.transforms :as transforms]))
 
+(defn- keycard-account?
+  [keypairs address]
+  (some (fn [keypair]
+          (->> keypair
+               :keycards
+               (some (fn [keycard]
+                       (-> keycard
+                           :accounts-addresses
+                           set
+                           (contains? address))))))
+        (vals keypairs)))
+
 (rf/reg-event-fx
- :wallet-connect/respond-current-session
+ :wallet-connect/authorized-signing
  (fn [{:keys [db]} [password]]
+   (let [prepared-hash (get-in db [:wallet-connect/current-request :prepared-hash])
+         address       (get-in db [:wallet-connect/current-request :address])
+         keycard-sign? (-> (get-in db [:wallet :keypairs])
```

Comment:
you need to move this outside, because there are two options
1) keypair is keycard we need to sign on keycard
2) profile is keycard, means we need to get password from keycard and sign regular way with password

Reply:
how can we know whether we need to actually sign on the keycard? I see that there's `(get-in transaction-for-signing [:signingDetails :signOnKeycard])`, but that comes from the router, which we don't use for dapps.

Reply:
`keycard-sign?` is correct you can use it, to know that you need to sign on keycard, but you need to move it outside of this event, you need to move this code 
```
(if keycard-sign?
       {:fx [[:dispatch
              [:standard-auth/authorize-with-keycard
              ```
              where :on-complete is used